tools/autograd/gen_python_functions.py

```diff
@@ -1359,13 +1359,13 @@ def make_python_binding_args(declaration):
         python_binding_arguments.append(dtype_arg)
 
     if is_factory_function or is_like_or_new_function_with_options:
-        py_default_layout = '*torch::getLayout(self.options().backend())' if is_like_or_new_function_with_options else None
+        py_default_layout = 'layout_from_backend(self.options().backend())' if is_like_or_new_function_with_options else None
         layout_arg = {
             'default': 'torch.strided',
-            'dynamic_type': 'Layout',
+            'dynamic_type': 'c10::optional<Layout>',
```

Comment:
optional or not here?

Reply:
@ezyang  quick question, what is the difference between dynamic_type, type, and simple_type?

Reply:
I'm not 100% sure. However, I think it's something like: `type` is the C++ type of the argument, complete with references. `simple_type` has all the references stripped off, and `optional` converted into the postfix question mark. And `dynamic_type` says what the actual runtime type is, e.g., `const Tensor&` might be the type, but `Tensor` is the `dynamic_type`. The best way to find out for sure, though, is to read the code responsible for generating these.